src/groups/mqb/mqbblp/mqbblp_relayqueueengine.cpp

```diff
@@ -130,6 +134,43 @@ class LimitedPrinter {
     LimitedPrinter& operator=(const LimitedPrinter&) BSLS_KEYWORD_DELETED;
 };
 
+struct Event {
+    const mqbi::QueueHandle* d_handle_p;
+    const QueueState*        d_queueState_p;
+    const bool               d_doesHandleExist;
+
+    explicit Event(const mqbi::QueueHandle* handle,
+                   const QueueState*        queueState,
+                   bool                     doesHandleExist)
+    : d_handle_p(handle)
+    , d_queueState_p(queueState)
+    , d_doesHandleExist(doesHandleExist)
+    {
+        // NOTHING
```

Comment:
```suggestion
        // PRECONDITIONS
        BSLS_ASSERT_SAFE(d_queueState_p);
        if (d_doesHandleExist) {
            BSLS_ASSERT_SAFE(d_handle_p);
        }
```
Is it possible to get rid of `d_doesHandleExist` at all? Can we look at `d_handle_p` instead?
What means to have a non-null `d_handle_p` if `d_doesHandleExist == false`?